<!DOCTYPE html>
<html lang = "en">
<link rel="stylesheet" href="style.css">

<head>
	<title>Karina Zadorozhny | Personal Website</title>
</head>


<body>
	<section id = 'navbar' class = 'container'>
		<div class="topnav">
			  <a href="./index.html"> Home</a>
			  <a href="./about.html"> About me </a>
			  <a class="active" href="./research.html">Projects</a>
			  <a href="./publications.html">Publications</a>
			  <a href="./connect.html">Connect</a>
		</div>
	</section>


	<section id ="research">
		<div class = "res">
			<h1 class = 'res-heading'> Projects </h1>
				<p class="res-text"> </p>

			<div class = 'res-project'>
			<h1 class = 'res-sub-heading'>1. Uncertainty in Neural Networks for Survival Predictions </h1>
				<p class="res-text"> 

					<p class = "res-text-affil">
						Insitute of Augmented Intelligence, Feinberg School of Medicine
					</p>

					<p class = "res-text-advisor">
						Lee Cooper Lab
					</p>

					<div class="res-image-right">
				        <img src="./images/projects3.png" alt="Uncertainty in Neural Networks" width="350">
     			 	</div>
					
					Understanding the certainty associated with a prediction isessential for translating machine-learning models into clinical practice. Uncertainty can help clinicians to better weight predictions in their decision-making processes, and can inform model developers about the deficiencies of models and where additional training data would be beneficial. 

					In this project, I have explored the use of Monte Carlo Dropout (MCD) as a method for estimating total uncertainty using MC methods applied at inference. While neural networks offer some advantages over Bayesian methods, MCD allows for approximation of the weights posterior.

					Using this approach, I have perfomed survival analysis on high-dimensional glioma samples and implemented the uncertainty estimation. The final model predicts the risk of individual patient as well as if we can trust this prediction, or rather, hand the case to a medical professional.


					<div class="res-image">
				        <img src="./images/projects4.png" alt="Uncertainty in Neural Networks" width=100%>
     			 	</div>

				</p>
			</div>

			<div class = 'res-project'>
			<h1 class = 'res-sub-heading'>2. Generative Adverserial Networks for Novel Protein Sequences </h1>
				<p class="res-text"> 

					<p class = "res-text-affil">
						Independent project
					</p>

					<div class="res-image-left">
					        <img src="./images/projects1.png" alt="Alpha-helix structure" width="350">
	     			 </div>
	     			 
	     			 Generative adversarial networks have recently been applied to protein and DNA sequence design. The models can be leveraged to create proteins with desired biochemical or functional properties such as presence of structural motifs or antimicrobial properties. While the previous work focused on optimizing for one property at a time, in this project, me and Lada Nuzhna aimed to use feedback GANs to design novel protein sequences that are simultaneously optimized for several properties. 

					The learning task is to train a GAN model such
					that it will be able to produce novel protein sequences enriched
					with one or more of 8 the possible structural properties.
					This was achieved by implementing a separate feedback
					network(s) that is grading generated sequences on
					the presence or absence of the structural motifs and updating
					the discriminator with the best performing fake sequences.



					<div class="res-image-center">
					        <img src="./images/projects2.png" alt="FB-GAN network" width=90%>
	     			 </div>

				</p>


			</div>

			<div class = 'res-project'>
			<h1 class = 'res-sub-heading'>3.  DNA Repair Mechanisms at Replication Forks </h1>

				<div class="res-image-right">
					        <img src="./images/projects5.jpg" alt="DNA repair at replication fork" width=120%>
	     			 </div>

				<p class="res-text"> 
						<p class = "res-text-affil">
							Biological Department, Faculty of Medicine, Masaryk University
						</p>
						<p class = "res-text-advisor">
							Lumir Krejci Lab
						</p>

					Fanconi anemia (FA) is a genetic disorder characterized by a defect in DNA interstrand crosslink (ICL) repair, chromosomal instability, and a predisposition to cancer. Recently, two RAD51 mutations were reported to cause an FA-like phenotype. Despite the tight association of FA/HR proteins with replication fork (RF) stabilization during normal replication, it remains unknown how FA-associated RAD51 mutations affect replication beyond ICL lesions. 
				</p>

				<p>		
					In this project, our laboratory has revealed that these mutations fail to protect nascent DNA from MRE11-mediated degradation during RF stalling in Xenopus laevis egg extracts. Reconstitution of DNA protection in vitro revealed that the defect arises directly due to altered RAD51 properties. Both mutations induce pronounced structural changes and RAD51 filament destabilization that is not rescued by prevention of ATP hydrolysis due to aberrant ATP binding. Our results further interconnect the FA pathway with DNA replication and provide mechanistic insight into the role of RAD51 in recombination-independent mechanisms of genome maintenance.
				</p>

				<p class="res-text"> 
					The main contributions of the works are the following:
					<ul class = 'res-list'>
					  <li>FA-associated RAD51 mutants are defective at stalled fork protection</li>
					  <li>Structural properties of the RAD51 nucleoprotein filament are altered by FA mutations</li>
					  <li>These filaments are highly unstable due to aberrant ATP binding and hydrolysis</li>
					</ul>

				</p>


			</div>

		</div>


	</section>


</body>

